pkg/webhook/cspc.go

```diff
@@ -503,3 +511,63 @@ func (wh *webhook) validateCSPCUpdateRequest(req *v1beta1.AdmissionRequest, getC
 
 	return response
 }
+
+// ValidateScaledown validates whether any cvr exist on the cspi
+// that is being scaled down
+func (p *PoolOperations) ValidateScaledown() (bool, string) {
+	removedPools := []string{}
+	for _, oldPool := range p.OldCSPC.Spec.Pools {
+		found := false
+		for _, newPool := range p.NewCSPC.Spec.Pools {
+			if reflect.DeepEqual(oldPool.NodeSelector, newPool.NodeSelector) {
+				found = true
+				break
+			}
+		}
+		if !found && p.IsScaledownCase(oldPool) {
+			cspi, err := p.clientset.CstorV1().CStorPoolInstances(p.OldCSPC.Namespace).
+				List(metav1.ListOptions{
+					LabelSelector: labels.SelectorFromSet(oldPool.NodeSelector).String() + "," +
+						types.CStorPoolClusterLabelKey + "=" + p.OldCSPC.Name,
+				})
+			if err != nil {
+				return false, fmt.Sprintf("Could not list cspi for cspc %s: %s", p.OldCSPC.Name, err.Error())
+			}
+			removedPools = append(removedPools, cspi.Items[0].Name)
+		}
+	}
+	for _, cspiName := range removedPools {
+		// list cvrs in all namespaces
+		cvrList, err := p.clientset.CstorV1().CStorVolumeReplicas("").List(metav1.ListOptions{
```

Comment:
no need to list in all-namespace here as pool is already exists, we can use the cspc `Namespace`.

Reply:
Since it is a single n/w call I believe there won't be any problem with that if the thought was about payload then we can do that.

Reply:
Done